server.go

```diff
@@ -185,6 +199,25 @@ func (p *peerState) forAllPeers(closure func(p *peer)) {
 	p.forAllOutboundPeers(closure)
 }
 
+// handleUpdatePeerHeight updates the heights of all peers who were known to
+// announce a block we recently accepted.
+func (s *server) handleUpdatePeerHeight(state *peerState, umsg updatePeerHeightMsg) {
+	state.forAllPeers(func(p *peer) {
```

Comment:
Why must we iterate all peers looking for block heights to update, when only a single peer needs this update?